### PR TITLE
Upgrade ADOT Collector version to v0.21.0

### PR DESCRIPTION
**Pull request by erichsueh3**
**Description:** <Describe what has changed.>
Upgrades ADOT Collector version to v0.21.0

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by vasireddy99 — APPROVED** [review]

File: docs/releases/v0.21.0.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+**Implemented enhancements:**
+
+- Export OTEL traces with Auth extension [\#1056](https://github.com/aws-observability/aws-otel-collector/issues/1056)
+
+**Merged pull requests:**
+
+- Update fargate deployment template to use include in cumulativetodelta processor [\#1435](https://github.com/aws-observability/aws-otel-collector/pull/1435) ([vasireddy99](https://github.com/vasireddy99))
+- Modify mirroring tool to use `crane` go pkg [\#1417](https://github.com/aws-observability/aws-otel-collector/pull/1417) ([erichsueh3](https://github.com/erichsueh3))
+- Update upstream collector dependancies to v0.58.0 [\#1414](https://github.com/aws-observability/aws-otel-collector/pull/1414) ([vasireddy99](https://github.com/vasireddy99))
+- Upgrade ADOT Collector version to v0.20.0 [\#1385](https://github.com/aws-observability/aws-otel-collector/pull/1385) ([Kausik-A](https://github.com/Kausik-A))
```
Comment (bryan-aguilar) [inline docs/releases/v0.21.0.md:21]:
```suggestion
```

**Review by bryan-aguilar — COMMENTED** [review]
I don't think we actively use it but can you update the multimod version also.

File: docs/releases/v0.21.0.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+[Full Changelog](https://github.com/aws-observability/aws-otel-collector/compare/v0.20.0...v0.21.0)
+
+**Upstream release notes:**
+
+- [v0.58.0](https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.58.0)
+- [v0.57.2](https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.57.2)
+
+**Implemented enhancements:**
+
+- Export OTEL traces with Auth extension [\#1056](https://github.com/aws-observability/aws-otel-collector/issues/1056)
```
Comment (Aneurysm9) [inline docs/releases/v0.21.0.md:14]: This seems to have been closed as stale.  Why is it listed as implemented?

File: docs/releases/v0.21.0.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+**Merged pull requests:**
+
+- Update fargate deployment template to use include in cumulativetodelta processor [\#1435](https://github.com/aws-observability/aws-otel-collector/pull/1435) ([vasireddy99](https://github.com/vasireddy99))
+- Modify mirroring tool to use `crane` go pkg [\#1417](https://github.com/aws-observability/aws-otel-collector/pull/1417) ([erichsueh3](https://github.com/erichsueh3))
+- Update upstream collector dependancies to v0.58.0 [\#1414](https://github.com/aws-observability/aws-otel-collector/pull/1414) ([vasireddy99](https://github.com/vasireddy99))
+- Upgrade ADOT Collector version to v0.20.0 [\#1385](https://github.com/aws-observability/aws-otel-collector/pull/1385) ([Kausik-A](https://github.com/Kausik-A))
+- Add operator arm64 tests [\#1365](https://github.com/aws-observability/aws-otel-collector/pull/1365) ([erichsueh3](https://github.com/erichsueh3))
```
Comment (Aneurysm9) [inline docs/releases/v0.21.0.md:22]: I'm not sure any of these are relevant to the end-user other than perhaps #1414.

**Review by vasireddy99 — APPROVED** [review]

File: docs/releases/v0.21.0.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# Changelog
+
+## [v0.21.0](https://github.com/aws-observability/aws-otel-collector/tree/v0.21.0) (2022-08-24)
```
Comment (Aneurysm9) [inline docs/releases/v0.21.0.md:3]: This isn't being released today, is it?

**Comment by vasireddy99** [comment]
looks like we should include the deprecation of `awsprometheusremotewriteexporter` in the changelog.

**Comment by Aneurysm9** [comment]
> looks like we should include the deprecation of `awsprometheusremotewriteexporter` in the changelog.

Not just deprecation, it's been eliminated, right?

**Comment by mhausenblas** [comment]
Yes we should call out `awsprometheusremotewriteexporter` and also, we should include migration guidance.

**Review by vasireddy99 — APPROVED** [review]

**Review by bryan-aguilar — APPROVED** [review]

**Pull request merged**